src/app/staticanalysis/modules/staticAnalysisCommon.js

```diff
@@ -17,7 +28,12 @@ var basicTypes = {
 
 var basicRegex = {
   CONTRACTTYPE: '^contract ',
-  FUNCTIONTYPE: '^function \\('
+  FUNCTIONTYPE: '^function \\(',
+  EXTERNALFUNCTIONTYPE: '^function \\(.*\\).* external',
+  CONSTANTFUNCTIONTYPE: '^function \\(.*\\).* constant',
+  REFTYPE: '( storage )|(mapping\\()|(\\[\\])',
```

Comment:
I include the spaces because somebody could name fields storage e.g. the type would be `struct KingOfTheEtherThrone.storage storage ref` without the spaces it would match the fields name. I don't have this problem with mappings because nobody can use parentheses in variable names.